gk/main.c

```diff
@@ -431,8 +550,43 @@ add_ggu_policy(struct ggu_policy *policy, struct gk_instance *instance)
 		fe->u.granted.budget_byte =
 			fe->u.granted.tx_rate_kb_cycle * 1024;
 
-		/* TODO Fill up the grantor id field. */
+		/* Find the saved lookup result. */
+		if (gt_id >= 0)
```

Comment:
Now that all states of a flow entry will know the grantor id, you can move the code that does the lookup for newly created entries all the way to the beginning of the function and leave this later part of the function to only deal with the transitions. This should make the code more readable.